tests/rptest/services/redpanda.py

```diff
@@ -1539,11 +1558,17 @@ def restart_nodes(self,
                       override_cfg_params=None,
                       start_timeout=None,
                       stop_timeout=None):
+
         nodes = [nodes] if isinstance(nodes, ClusterNode) else nodes
-        for node in nodes:
-            self.stop_node(node, timeout=stop_timeout)
-        for node in nodes:
-            self.start_node(node, override_cfg_params, timeout=start_timeout)
+        with concurrent.futures.ThreadPoolExecutor(
+                max_workers=len(nodes)) as executor:
+            list(
+                executor.map(lambda n: self.stop_node(n, timeout=stop_timeout),
```

Comment:
what will happen if executor thread throws ? i am wondering if we would lose timeout exception in this case ?

Reply:
From map() docs:
> If a func call raises an exception, then that exception will be raised when its value is retrieved from the iterator.

The call to `list()` is the trick here: it pumps the iterator and makes sure that if any of the results were exceptional they are raised.

Reply:
perfect